NuRadioReco/modules/io/coreas/readCoREASDetector.py

```diff
@@ -0,0 +1,222 @@
+    -------
+    channel_ids : defaultdict
+        Dictionary with channel group ids as keys and lists of channel ids as values
+    """
+    channel_ids = defaultdict(list)
+    for channel_id in requested_channel_ids:
+        if channel_id in det.get_channel_ids(station_id):
+            channel_group_id = det.get_channel_group_id(station_id, channel_id)
+            channel_ids[channel_group_id].append(channel_id)
+    return channel_ids
+
+
+class readCoREASDetector():
+    """
+    Use this as default when reading CoREAS files and combining them with a detector.
+
+    This model reads the electric fields of a CoREAS file with a star shaped pattern and foldes them with it given detector. 
+    The electric field of the star shaped pattern is interpolated at the detector positions. If the angle between magnetic field 
+    and shower direction are below about 15 deg, the interpolation is no longer reliable and the closest observer is used instead.
+    """
+
+    def __init__(self):
+        self.__t = 0
+        self.__t_event_structure = 0
+        self.__t_per_event = 0
+        self.__corsika_evt = None
+        self.__interp_lowfreq = None
+        self.__interp_highfreq = None
+        self.logger = logging.getLogger('NuRadioReco.readCoREASDetector')
+
+    def begin(self, input_file, interp_lowfreq=30*units.MHz, interp_highfreq=1000*units.MHz, log_level=logging.INFO):
+        """
+        begin method, initialize readCoREASDetector module
+
+        Parameters
+        ----------
+        input_files: str
+            coreas hdf5 file
+        interp_lowfreq: float (default = 30 MHz)
+            lower frequency for the bandpass filter in interpolation, should be broader than the sensetivity band of the detector
+        interp_highfreq: float (default = 1000 MHz)
+            higher frequency for the bandpass filter in interpolation,  should be broader than the sensetivity band of the detector
+        """
+        self.logger.setLevel(log_level)
+        filesize = os.path.getsize(input_file)
+        if(filesize < 18456 * 2):  # based on the observation that a file with such a small filesize is corrupt
+            self.logger.warning("file {} seems to be corrupt".format(input_file))
+        self.__corsika_evt = coreas.read_CORSIKA7(input_file)
+        self.logger.info(f"using coreas simulation {input_file} with E={self.__corsika_evt.get_first_sim_shower().get_parameter(shp.energy):.2g}eV, zenith angle = {self.__corsika_evt.get_first_sim_shower().get_parameter(shp.zenith) / units.deg:.0f}deg")
+
+        self.__interp_lowfreq = interp_lowfreq
+        self.__interp_highfreq = interp_highfreq
+
+        self.coreasInterpolator = coreas.coreasInterpolator(self.__corsika_evt)
+        self.coreasInterpolator.initialize_efield_interpolator(self.__interp_lowfreq, self.__interp_highfreq)
+
+    @register_run()
+    def run(self, detector, core_position_list=[], selected_station_ids=[], selected_channel_ids=[]):
+        """
```

Comment:
I think it is also important to mention here that the traces are smoothed with the half-Hann window by default. If one uses shorter traces, then this might have a big impact.